cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/SupportBundleHttpHandler.java

```diff
@@ -0,0 +1,610 @@
+        Constants.AppFabric.QUERY_PARAM_START_TIME,
+        Collections.singletonList(String.valueOf(startTs - 5000)));
+    queryParams.put(
+        Constants.AppFabric.QUERY_PARAM_END_TIME,
+        Collections.singletonList(String.valueOf(stopTs)));
+    MetricQueryResult metricQueryResult =
+        metricsQueryHelper.executeTagQuery(queryTags, metricsList, new ArrayList<>(), queryParams);
+    JsonObject metrics = new JsonObject();
+    for (MetricQueryResult.TimeSeries timeSeries : metricQueryResult.getSeries()) {
+      if (!metrics.has(timeSeries.getMetricName())) {
+        metrics.add(timeSeries.getMetricName(), new JsonArray());
+      }
+      for (MetricQueryResult.TimeValue timeValue : timeSeries.getData()) {
+        JsonObject time = new JsonObject();
+        time.addProperty("time", timeValue.getTime());
+        time.addProperty("value", timeValue.getValue());
+        metrics.getAsJsonArray(timeSeries.getMetricName()).add(time);
+      }
+    }
+    return metrics;
+  }
+
+  // Zips all the file under support bundle directory */
+  //  private void zipFile(String dirPath) {
+  //    try {
+  //      File zipFile = new File("/Users/bjzhu/Documents/bjzhu/collect_support_bundle.zip");
+  //      File confDir = new File(dirPath);
+  //      try (ZipOutputStream zipOutput = new ZipOutputStream(new FileOutputStream(zipFile))) {
+  //        BundleJarUtil.addToArchive(confDir, zipOutput);
+  //      }
+  //    } catch (IOException e) {
+  //      throw new RuntimeException(e);
+  //    }
+  //  }
+
+  /** Generates pipeline log file and run info file */
+  private void generateLogFileAndRunInfo(
+      ProgramRunId latestProgramRunId,
+      RunRecordDetail latestRunRecordDetail,
+      Boolean needLog,
+      String appPath,
+      JsonObject metrics,
+      List<Thread> threadList,
+      FileWriter statusJson) {
+    String programType = "workflows";
+
+    if (needLog) {
+      // Generates pipeline log file for user request
+      final Thread thread =
+          new Thread(
+              new Runnable() {
+                @Override
+                public void run() {
+                  try {
+                    ProgramType type = ProgramType.valueOfCategoryName(programType);
+                    ProgramRunId programRunId =
+                        new ProgramRunId(
+                            latestProgramRunId.getNamespace(),
+                            latestProgramRunId.getApplication(),
+                            type,
+                            latestProgramRunId.getProgram(),
+                            latestProgramRunId.getRun());
+                    LoggingContext loggingContext =
+                        LoggingContextHelper.getLoggingContextWithRunId(latestProgramRunId, null);
+                    generateLogFile(
+                        loggingContext, appPath, latestProgramRunId.getRun() + "-log.txt");
+                  } catch (Exception e) {
+                    e.printStackTrace();
```

Comment:
Please do proper exception logging with a logger and also write any problems you encounter to a respective file if possible